lib/internal/Magento/Framework/View/Layout/GeneratorPool.php

```diff
@@ -86,11 +87,17 @@ public function process(Reader\Context $readerContext, Generator\Context $genera
      * Add generators to pool
      *
      * @param GeneratorInterface[] $generators
+     *
      * @return void
      */
     protected function addGenerators(array $generators)
     {
         foreach ($generators as $generator) {
+            if (!$generator instanceof GeneratorInterface) {
+                throw new \InvalidArgumentException(
+                    'Generator class must be an instance of %1', GeneratorInterface::class
```

Comment:
Yep, better use concatenation here (looks like you intended to use `sprintf`).

Commit amend + force push, please.

Reply:
@orlangur I'm just using `sprintf()` now.